### ISSUE TITLE
dictionary_sugar_get seems to fail when dictionary entries are functions

### ISSUE THREAD

**Issue by mb706** [mlr-org/mlr3misc#79]
```r
library("mlr3")
rsmp("holdout", x = 1)
#> Error: object of type 'closure' is not subsettable
#> Environment:
#>   1: rsmp("holdout", x = 1)
#>   2: dictionary_sugar_get(mlr_resamplings, .key, ...)
#>   3: stopf("Cannot set argument '%s' for '%s' (not a constructor argument, not a
#>   4: stop(simpleError(str_wrap(sprintf(msg, ...), width = wrap), call = NULL))
#>   5: simpleError(str_wrap(sprintf(msg, ...), width = wrap), call = NULL)
#>   6: structure(list(message = as.character(message), call = call), class = class
#>   7: str_wrap(sprintf(msg, ...), width = wrap)
#>   8: sprintf(msg, ...)
#>   9: did_you_mean(nn, c(constructor_args, param_ids, fields(obj$value)))
#>   10: unique(candidates)
#>   11: fields(obj$value)
#>   12: setdiff(names(x$public_methods), c("initialize", "clone", "print", "format"
#>   13: as.vector(x)
```
problem seems to be that `rsmp()`-entries are functions and not `R6`-constructors, which is explicitly allowed for dictionaries.

**Comment by mb706** [comment]
(expected behaviour: `"Error Cannot set argument 'x' for 'holdout' ..."`)

**Comment by advieser** [comment]
I think this is the same as https://github.com/mlr-org/mlr3misc/issues/115 and was fixed in https://github.com/mlr-org/mlr3misc/pull/116, so it could be closed, @sebfischer? (since you made the fix)

**Comment by mb706** [comment]
Yes, fixed by #116
